Panic Error

Hello,
using latest version I'm receiving

panic: interface conversion: interface {} is nil, not string

goroutine 82 [running]:
github.com/Azure/azqr/internal.AprlScanner.graphScan({}, {0x1fd3f68, 0xc0003c8050}, 0xc000386210, {0xc0003257c0, 0xc, 0x0?}, 0xc00048ae40)
        D:/a/azqr/azqr/internal/aprl_scanner.go:245 +0xc2a
github.com/Azure/azqr/internal.(*AprlScanner).worker(0x2c76f60, {0x1fd3f68, 0xc0003c8050}, 0xc000386210, 0xc00048ae40, 0xc0004b24d0, 0xc0004b2540, 0xc0005b6c50)
        D:/a/azqr/azqr/internal/aprl_scanner.go:175 +0x127
created by github.com/Azure/azqr/internal.AprlScanner.Scan in goroutine 1
        D:/a/azqr/azqr/internal/aprl_scanner.go:138 +0x1ce

Regards,
Red.

Hey there,
here we are:

2024-10-31T15:06:57+01:00 DBG // Azure Resource Graph query
// Public IP addresses should have DDoS protection enabled
resources
| where type =~ 'Microsoft.Network/publicIPAddresses'
| where properties.ddosSettings.protectionMode !in~ ("Enabled", "VirtualNetworkInherited")
| project recommendationId="c4254c66-b8a5-47aa-82f6-e7d7fb418f47", name, id, tags, param1=strcat("Apply either DDoS Network protection or DDoS IP Protrection to the public IP address.")

panic: interface conversion: interface {} is nil, not string

goroutine 51 [running]:
github.com/Azure/azqr/internal.AprlScanner.graphScan({}, {0x1fd3f68, 0xc000097770}, 0xc00007a1d8, {0xc00035f7c0, 0xc, 0x0?}, 0xc00027b410)
        D:/a/azqr/azqr/internal/aprl_scanner.go:245 +0xc2a
github.com/Azure/azqr/internal.(*AprlScanner).worker(0x2c76f60, {0x1fd3f68, 0xc000097770}, 0xc00007a1d8, 0xc00027b410, 0xc000300930, 0xc0003009a0, 0xc000503880)
        D:/a/azqr/azqr/internal/aprl_scanner.go:175 +0x127
created by github.com/Azure/azqr/internal.AprlScanner.Scan in goroutine 1
        D:/a/azqr/azqr/internal/aprl_scanner.go:138 +0x1ce

Regards,
Red.

Hello,
the tool worked for a long time tha I received:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xf8e8a5]

goroutine 17643 [running]:
github.com/Azure/azqr/internal/scanners/pip.(*PublicIPScanner).Scan(0xc0003882d0, 0xc02e4bad80)
        D:/a/azqr/azqr/internal/scanners/pip/pip.go:45 +0x225
github.com/Azure/azqr/internal.Scanner.retry({}, 0x3, 0x0?, {0x1bd3a48, 0xc0003882d0}, 0xc02e4bad80)
        D:/a/azqr/azqr/internal/scanner.go:237 +0x103
github.com/Azure/azqr/internal.Scanner.Scan.func1({0x1bd3a48?, 0xc0003882d0?})
        D:/a/azqr/azqr/internal/scanner.go:181 +0x5c
created by github.com/Azure/azqr/internal.Scanner.Scan in goroutine 1
        D:/a/azqr/azqr/internal/scanner.go:180 +0xc87

I'll run it again with debug option

Regards,
Red.

--- a/internal/aprl_scanner.go
+++ b/internal/aprl_scanner.go
@@ -196,36 +196,6 @@ func (sc AprlScanner) graphScan(ctx context.Context, graphClient *graph.GraphQue
 				for _, row := range result.Data {
 					m := row.(map[string]interface{})
 
-					tags := ""
-					if m["tags"] != nil {
-						tags = convertInterfaceToString(m["tags"])
-					}
-
-					param1 := ""
-					if m["param1"] != nil {
-						param1 = convertInterfaceToString(m["param1"])
-					}
-
-					param2 := ""
-					if m["param2"] != nil {
-						param2 = convertInterfaceToString(m["param2"])
-					}
-
-					param3 := ""
-					if m["param3"] != nil {
-						param3 = convertInterfaceToString(m["param3"])
-					}
-
-					param4 := ""
-					if m["param4"] != nil {
-						param4 = convertInterfaceToString(m["param4"])
-					}
-
-					param5 := ""
-					if m["param5"] != nil {
-						param5 = convertInterfaceToString(m["param5"])
-					}
-
 					log.Debug().Msg(rule.GraphQuery)
 
 					subscription := azqr.GetSubsctiptionFromResourceID(m["id"].(string))
@@ -242,17 +212,17 @@ func (sc AprlScanner) graphScan(ctx context.Context, graphClient *graph.GraphQue
 						LongDescription:     rule.LongDescription,
 						PotentialBenefits:   rule.PotentialBenefits,
 						Impact:              azqr.RecommendationImpact(rule.Impact),
-						Name:                m["name"].(string),
-						ResourceID:          m["id"].(string),
+						Name:                convertInterfaceToString(m["name"]),
+						ResourceID:          convertInterfaceToString(m["id"]),
 						SubscriptionID:      subscription,
 						SubscriptionName:    subscriptionName,
 						ResourceGroup:       azqr.GetResourceGroupFromResourceID(m["id"].(string)),
-						Tags:                tags,
-						Param1:              param1,
-						Param2:              param2,
-						Param3:              param3,
-						Param4:              param4,
-						Param5:              param5,
+						Tags:                convertInterfaceToString(m["tags"]),
+						Param1:              convertInterfaceToString(m["param1"]),
+						Param2:              convertInterfaceToString(m["param2"]),
+						Param3:              convertInterfaceToString(m["param3"]),
+						Param4:              convertInterfaceToString(m["param4"]),
+						Param5:              convertInterfaceToString(m["param5"]),
 						Learn:               rule.LearnMoreLink[0].Url,
 						AutomationAvailable: rule.AutomationAvailable,
 						Source:              "APRL",
@@ -289,6 +259,10 @@ func (sc AprlScanner) getGraphRules(service string, filters *azqr.Filters, aprl 
 }
 
 func convertInterfaceToString(i interface{}) string {
+	if i == nil {
+		return ""
+	}
+
 	switch v := i.(type) {
 	case string:
 		return v

